src/test/java/io/cdap/plugin/run/transform/RunTest.java

```diff
@@ -115,7 +142,7 @@ public void testRunWithJarInput() throws Exception {
     MockSource.writeInput(inputManager, input);
     WorkflowManager workflowManager = appManager.getWorkflowManager(SmartWorkflow.NAME);
     workflowManager.start();
-    workflowManager.waitForStopped(5, TimeUnit.MINUTES);
+    workflowManager.waitForRun(ProgramRunStatus.COMPLETED, 5, TimeUnit.MINUTES);
```

Comment:
start() followed by waitForRun() should be replaced by startAndWaitForRun().

waitForRun() is a confusing method that is easily misused if the same program is run more than once. It simple checks that there is at least one run record in the given state. So if the program already ran once successfully before this call, this will return immediately.

Reply:
Done.